File: backend/src/main/java/com/board/backend/service/BoardService.java
package com.board.backend.service;

import com.board.backend.model.BoardDto;

public interface BoardService {

    BoardDto.ResponseList list(BoardDto.RequestList form);
    BoardDto.Response get(long no, boolean incrementViewCount);
    BoardDto.Response insert(BoardDto.Create form);
    BoardDto.Response update(long no, BoardDto.Update form);
    BoardDto.Response delete(long no);
}

File: frontend/src/common/services/BoardService.js
import { AppConstants } from '../index'
import HttpClient from './HttpClient'

const SERVICE_URL = `${AppConstants.BASE_URL}/board`

const BoardService = {
  list: async (form) => {
    const url = `${SERVICE_URL}/list`
    return await HttpClient.get(url, form)
  },
  get: async (no) => {
    const url = `${SERVICE_URL}/${no}`
    return await HttpClient.get(url)
  },
  create: async (form) => {
    const url = SERVICE_URL
    return await HttpClient.post(url, form)
  },
  update: async (no, form) => {
    const url = `${SERVICE_URL}/${no}`
    return await HttpClient.put(url, form)
  },
  delete: async (no) => {
    const url = `${SERVICE_URL}/${no}`
    return await HttpClient.delete(url)
  },
}

export default BoardService
